### PR TITLE
Content unread fix

### PR DESCRIPTION
**Pull request by mikkpokk**

### CODE REVIEW

**Comment by mikkpokk** [comment]
Before merging do not forget to modify variable "except_timestamp_to"

File: app/Content.php [hunk 1]
```diff
@@ -17,10 +17,32 @@ class Content extends Model
 
     protected $dates = ['created_at', 'updated_at', 'start_at', 'end_at'];
 
-    protected $appends = ['body_filtered', 'image_id'];
+    protected $appends = ['body_filtered', 'image_id', 'unread', 'first_unread_comment_id'];
 
     // Relations
 
+    public function getUnreadAttribute()
```
Comment (kristjanjansen) [inline app/Content.php]: Why not keep it all at V2ContentVars? We moved all magical computed vars and attributes out behind presenter class, accessible via ->vars(), keeping only db-based attributes and relations in model php file. We could revisit that later (bring that stuff in via traits etc) but ->vars() is what we standardized for now.

File: app/Http/Controllers/CommentController.php [hunk 2]
```diff
@@ -60,27 +58,6 @@ public function store(Request $request, $type, $content_id)
         }
 
         */
-        if (in_array($comment->content->type, ['forum', 'buysell', 'expat', 'internal'])) {
```
Comment (kristjanjansen) [inline app/Http/Controllers/CommentController.php]: Love this hodgepodge to go :)

File: app/Http/Regions/ForumRow.php [hunk 3]
```diff
@@ -8,8 +8,10 @@ public function render($forum, $route = '')
     {
         $user = request()->user();
         $commentCount = $forum->vars()->commentCount;
-        $unreadCommentCount = $forum->vars()->unreadCommentCount;
-        $firstUnreadCommentId = $forum->vars()->firstUnreadCommentId;
+        //$unreadCommentCount = $forum->vars()->unreadCommentCount;
+        $unreadCommentCount = $forum->unread;
```
Comment (kristjanjansen) [inline app/Http/Regions/ForumRow.php]: Lets keep `$forum->vars()->unreadCommentCount` as a standard naming convention. You could do it in followup of course.

File: app/UnreadContent.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class UnreadContent extends Model
+{
+    public $timestamps = false;
+    protected static $except_timestamp_to = '2017-07-14 14:00:00'; // todo
```
Comment (kristjanjansen) [inline app/UnreadContent.php]: Why?

File: app/UnreadContent.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    public function content()
+    {
+        return $this->hasOne('App\Content', 'id', 'content_id');
+    }
+
+    public function getUnread(Content $content)
+    {
+        $unread_timestamp = strtotime($this->read_at);
```
Comment (kristjanjansen) [inline app/UnreadContent.php]: Do not get the `strtotime`stuff. Could we not use Carbon here for all date wrangling and Eloquent date casting?

File: app/UnreadContent.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        $unread_content = $content->unread_content;
+
+        $unread_data = [
+            'count' => 0,
+            'first_comment_id' => null,
+        ];
+
+        if ($unread_content && auth()->check()) {
+            return (int) $unread_content->getUnread($content);
```
Comment (kristjanjansen) [inline app/UnreadContent.php]: Add a comment about casting

File: app/UnreadContent.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        } elseif (auth()->check()) {
+            $except_timestamp = strtotime(self::$except_timestamp_to);
```
Comment (kristjanjansen) [inline app/UnreadContent.php]: do not get `except_timestamp`, I assume for debug / testing?

File: app/UnreadContent.php (hunk 4 above)
Comment (mikkpokk) [inline app/UnreadContent.php]: Otherwise after merging every posts are "unread"

**Pull request merged**